### DISCUSSION TITLE
Previews before sending?

### DISCUSSION THREAD

**Discussion by selinosherpa** [rpldy/react-uploady#239] [Q&A]
Is it possible to select images, show the previews, and then trigger the upload?

**Comment by yoavniran** [comment]
Currently the Uploady [upload-preview](https://github.com/rpldy/react-uploady/tree/master/packages/ui/upload-preview) package is using the Batch Start event so it wont show previews for files added with autoUpload = false.

However, you could write your own preview component using the Batch Add (BATCH_ADD) event/[hook](https://react-uploady.netlify.app/docs/api/hooks/useBatchAddListener/).

If there's a need it may be possible in the future to add a flag for the preview to use Batch Add for this kind of scenario

**Reply by yoavniran** [reply]
It is possible now to pass a different method to intercept the batch-add event. see: [Custom Batch Items Method](https://react-uploady.netlify.app/docs/api/utils/getUploadPreviewForBatchItemsMethod/)
